test/e2e/tests/resize_e2e_test.go

```diff
@@ -0,0 +1,112 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/gcp-filestore-csi-driver/pkg/util"
+	testutils "sigs.k8s.io/gcp-filestore-csi-driver/test/e2e/utils"
+)
+
+var _ = Describe("GCE Filestore CSI Driver", func() {
+	It("Should online resize node for volume", func() {
+		testContext := getRandomTestContext()
+
+		client := testContext.Client
+		instance := testContext.Instance
+
+		// Create Disk
+		volName := testNamePrefix + string(uuid.NewUUID())
+		vol, err := client.CreateVolume(volName)
+		Expect(err).To(BeNil(), "CreateVolume failed with error: %v", err)
+
+		defer func() {
+			// Delete Disk
+			client.DeleteVolume(vol.GetVolumeId())
+			Expect(err).To(BeNil(), "DeleteVolume failed")
+
+			_, err := getDisk(volName, instance)
+			Expect(err).NotTo(BeNil(), "Could get deleted disk from cloud directly")
+		}()
+
+		// Validate Disk Created
+		inst, err := getDisk(volName, instance)
+		Expect(err).To(BeNil(), "Could not get disk from cloud directly")
+		Expect(inst.State).To(Equal(readyState))
+		Expect(inst.Tier).To(Equal(defaultTier))
+		Expect(inst.Networks[0].Network).To(Equal(defaultNetwork))
+		Expect(inst.FileShares[0].CapacityGb).To(Equal(util.RoundBytesToGb(minVolumeSize)))
+
+		// Stage Disk
+		stageDir := filepath.Join("/tmp/", volName, "stage")
+		err = testutils.MkdirAll(instance, stageDir)
+		Expect(err).To(BeNil(), "Mkdir failed with error")
+
+		err = client.NodeStageVolume(vol.GetVolumeId(), stageDir, vol.GetVolumeContext())
+		Expect(err).To(BeNil(), "NodeStageVolume failed with error")
+		defer func() {
+			// Unstage Disk
+			err = client.NodeUnstageVolume(vol.GetVolumeId(), stageDir)
+			Expect(err).To(BeNil(), "NodeUnstageVolume failed with error")
+
+			// delete remote directory
+			fp := filepath.Join("/tmp/", volName)
+			err = testutils.RmAll(instance, fp)
+			Expect(err).To(BeNil(), "Failed to delete remote directory")
+		}()
+
+		// Mount Disk
+		publishDir := filepath.Join("/tmp/", volName, "mount")
+		// make remote directory
+		_, err = instance.SSH(fmt.Sprint("mkdir -p ", publishDir))
+		Expect(err).To(BeNil(), "Failed to create remote directory")
+
+		err = client.NodePublishVolume(vol.GetVolumeId(), stageDir, publishDir, vol.GetVolumeContext())
+		Expect(err).To(BeNil(), "NodePublishVolume failed with error")
+		defer func() {
+			// Unmount Disk
+			err = client.NodeUnpublishVolume(vol.GetVolumeId(), publishDir)
+			Expect(err).To(BeNil(), "NodeUnpublishVolume failed with error")
+		}()
+
+		// Verify pre-resized disk size
+		sizeTb, err := testutils.GetFSSizeInTb(instance, publishDir)
+		Expect(err).To(BeNil(), "Failed to get FSSize in TB")
+		Expect(sizeTb * util.Tb).To(Equal(minVolumeSize))
+
+		// Resize controller
```

Comment:
Few comments:
a) I think it will be good to create a function of this:
    1. resize api call 2. verification of resized instances.
That way we can add all cases in a single place and reuse in both online and offline test case.

b) Add resize call on the same instance with limit bytes set in the resize csi request. (it can be one simple case, where both required and limit bytes are set in the request. for exhaustive checks of limit and required bytes we can add to UTs if needed). Thoughts?
c) Add a resize call which increments in the order of bytes (because the doc says that https://cloud.google.com/filestore/docs/editing-instances#differences_in_scaling_behavior_by_service_tier basic tier can be incremented in the order of GBs)
d) Add a resize call with lesser than the current size of the instance. not sure what the API returns, but I expect it to fail. basic HDD and basic SSD can only be expanded.

All of resize calls, b c d can be done on the same instance.

Reply:
Refactored tests to act on single instance. Added b/c/d to current implementation. Lmk your thoughts of current implementation w.r.t a).

Reply:
Added timing log within client-wrapper in commit 1.